### PR TITLE
fixed https://github.com/NuGet/Home/issues/1099

### PR DESCRIPTION
**Pull request by zhili1208**
https://github.com/NuGet/Home/issues/1099
the bug is when UI try to get metadata from each source, if one bad source is before good source, it will just throw, never try good source.

the fix is catch exception for each source when it throw, if there is no good source, throw all exceptions. otherwise, if there is one good source, just return the metadata

### CODE REVIEW

File: src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs [hunk 1]
```diff
@@ -376,16 +378,28 @@ protected static IReadOnlyList<PreviewResult> GetPreviewResults(IEnumerable<Tupl
...
+                        includePrerelease: true,
+                        includeUnlisted: true,
+                        token: token);
+                    var packageMetadata = r.FirstOrDefault(p => p.Identity.Version == package.Version);
+                    if (packageMetadata != null)
+                    {
+                        return packageMetadata;
+                    }
                 }
+                catch(InvalidOperationException e)
```
Comment (emgarten) [inline src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs:394]: Wouldn't this be throwing a ProtocolException?

Comment (zhili1208) [inline src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs:394]: no, this throwed InvalidOperationException

**Comment by emgarten** [comment]
:watch: UserAction should carry the UIPackageMetadata which already has this info. See if that is possible. It would save an extra http call.

**Comment by emgarten** [comment]
:shipit:

File: src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs (hunk 1 above)
Comment (deepakaravindr) [inline src/NuGet.Clients/PackageManagement.UI/Actions/UIActionEngine.cs:394]: Perhaps, we should also catch ProtocolException, unless, you are absolutely sure that this method cannot throw any other exception

**Comment by deepakaravindr** [comment]
:shipit:

**Pull request closed without merging**